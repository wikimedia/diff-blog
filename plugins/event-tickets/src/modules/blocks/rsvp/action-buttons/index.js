export { default as AttendeesActionButton } from './attendees-action-button/container';
export { default as SettingsActionButton } from './settings-action-button/container';
