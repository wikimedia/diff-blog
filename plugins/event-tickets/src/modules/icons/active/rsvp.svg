<svg width="60" height="60" xmlns="http://www.w3.org/2000/svg"><g fill="none" fill-rule="evenodd"><path d="M5.419 24.046l26.093 9.307 16.183-22.117.567.858-15.905 25.354a1.926 1.926 0 0 1-2.491.701L5.012 25.761l.407-1.715z" fill="#E6E6E6"/><path fill="#039ED3" d="M12.517 49.828l-1.762-.945 10.58-19.731 1.763.946zM55.309 37.557l-19.43-11.125.995-1.736 19.429 11.125z"/><path d="M3 9h54.017v41.354H3z"/><path d="M5.237 23.411l7.084 24.706L54.78 35.942l-7.084-24.706L5.237 23.411zm6.398 26.943a1.005 1.005 0 0 1-.961-.724L3.04 23.001a1 1 0 0 1 .685-1.237L48.105 9.038a1.002 1.002 0 0 1 1.237.685l7.635 26.63a1 1 0 0 1-.685 1.236l-44.38 12.726a.959.959 0 0 1-.277.04z" fill="#444"/><path d="M31.1 34.48c-.118 0-.237-.02-.352-.063L3.83 24.295l.704-1.871 26.222 9.86 17.012-22.257 1.59 1.215-17.463 22.846a1 1 0 0 1-.795.393" fill="#444"/></g></svg>