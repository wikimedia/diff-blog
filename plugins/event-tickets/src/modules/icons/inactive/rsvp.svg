<svg width="60" height="60" xmlns="http://www.w3.org/2000/svg"><g fill="#AEB4BB" fill-rule="evenodd"><path d="M12.517 49.828l-1.762-.945 10.58-19.731 1.763.946zM55.309 37.557l-19.43-11.125.995-1.736 19.429 11.125z"/><path d="M5.237 23.411l7.084 24.706L54.78 35.942l-7.084-24.706L5.237 23.411zm6.398 26.943a1.005 1.005 0 0 1-.961-.724L3.04 23.001a1 1 0 0 1 .685-1.237L48.105 9.038a1.002 1.002 0 0 1 1.237.685l7.635 26.63a1 1 0 0 1-.685 1.236l-44.38 12.726a.959.959 0 0 1-.277.04z"/><path d="M31.1 34.48c-.118 0-.237-.02-.352-.063L3.83 24.295l.704-1.871 26.222 9.86 17.012-22.257 1.59 1.215-17.463 22.846a1 1 0 0 1-.795.393"/></g></svg>