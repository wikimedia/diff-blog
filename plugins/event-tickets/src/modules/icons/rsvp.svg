<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 47.99 38.9"><defs><style>.cls-1{fill:#010101;}</style></defs><title>block-icon-rsvp</title><g id="Layer_2" data-name="Layer 2"><g id="Layer_1-2" data-name="Layer 1"><path class="cls-1" d="M47.93,28l-5-26.27L26.1,27a1.11,1.11,0,0,1-1.57.3L0,10.38,5,36.29a3.21,3.21,0,0,0,3.75,2.54l36.67-7A3.21,3.21,0,0,0,47.93,28Z"/><path class="cls-1" d="M23.64,21.23a.81.81,0,0,0,1-.11L39.93,0,1.65,7.34,22.49,20.51Z"/></g></g></svg>