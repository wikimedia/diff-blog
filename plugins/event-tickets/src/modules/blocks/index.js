/**
 * Wordpress dependencies
 */
import { registerBlockType } from '@wordpress/blocks';

/**
 * Internal dependencies
 */
import { initStore } from '@moderntribe/tickets/data';
import rsvp from '@moderntribe/tickets/blocks/rsvp';
import tickets from '@moderntribe/tickets/blocks/tickets';
import ticket from '@moderntribe/tickets/blocks/ticket';
import attendees from '@moderntribe/tickets/blocks/attendees';

const blocks = [
	rsvp,
	tickets,
	ticket,
	attendees,
];

blocks.forEach( ( block ) => registerBlockType( `tribe/${ block.id }`, block ) );

// Initialize AFTER blocks are registered
// to avoid plugin shown as available in reducer
// but not having block available for use
initStore();

export default blocks;
