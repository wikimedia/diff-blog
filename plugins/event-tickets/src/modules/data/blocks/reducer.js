/**
 * External dependencies
 */
import { combineReducers } from 'redux';

/**
 * Internal dependencies
 */
import rsvp from './rsvp';
import ticket from './ticket';
import attendees from './attendees';

export default combineReducers( {
	rsvp,
	ticket,
	attendees,
} );
