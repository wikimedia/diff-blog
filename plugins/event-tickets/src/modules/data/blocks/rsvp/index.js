/**
 * Internal dependencies
 */
import * as types from './types';
import * as actions from './actions';
import * as selectors from './selectors';
import * as thunks from './thunks';
import reducer from './reducer';
import sagas from './sagas';

export default reducer;
export { types, actions, sagas, selectors, thunks };
