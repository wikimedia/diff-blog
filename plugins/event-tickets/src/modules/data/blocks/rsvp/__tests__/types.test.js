/**
 * Internal dependencies
 */
import { types } from '@moderntribe/tickets/data/blocks/rsvp';

describe( 'RSVP block types', () => {
	const keys = Object.keys( types );

	keys.forEach( ( key ) => {
		test( key, () => {
			expect( types[ key ] ).toMatchSnapshot();
		} );
	} );
} );
