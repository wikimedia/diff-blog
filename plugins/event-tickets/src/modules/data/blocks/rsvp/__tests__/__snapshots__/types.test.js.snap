// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RSVP block types CREATE_RSVP 1`] = `"@@MT/TICKETS/CREATE_RSVP"`;

exports[`RSVP block types DELETE_RSVP 1`] = `"@@MT/TICKETS/DELETE_RSVP"`;

exports[`RSVP block types DELETE_RSVP_HEADER_IMAGE 1`] = `"@@MT/TICKETS/DELETE_RSVP_HEADER_IMAGE"`;

exports[`RSVP block types FETCH_RSVP_HEADER_IMAGE 1`] = `"@@MT/TICKETS/FETCH_RSVP_HEADER_IMAGE"`;

exports[`RSVP block types HANDLE_RSVP_END_DATE 1`] = `"@@MT/TICKETS/HANDLE_RSVP_END_DATE"`;

exports[`RSVP block types HANDLE_RSVP_END_TIME 1`] = `"@@MT/TICKETS/HANDLE_RSVP_END_TIME"`;

exports[`RSVP block types HANDLE_RSVP_START_DATE 1`] = `"@@MT/TICKETS/HANDLE_RSVP_START_DATE"`;

exports[`RSVP block types HANDLE_RSVP_START_TIME 1`] = `"@@MT/TICKETS/HANDLE_RSVP_START_TIME"`;

exports[`RSVP block types INITIALIZE_RSVP 1`] = `"@@MT/TICKETS/INITIALIZE_RSVP"`;

exports[`RSVP block types SET_RSVP_CAPACITY 1`] = `"@@MT/TICKETS/SET_RSVP_CAPACITY"`;

exports[`RSVP block types SET_RSVP_DESCRIPTION 1`] = `"@@MT/TICKETS/SET_RSVP_DESCRIPTION"`;

exports[`RSVP block types SET_RSVP_DETAILS 1`] = `"@@MT/TICKETS/SET_RSVP_DETAILS"`;

exports[`RSVP block types SET_RSVP_END_DATE 1`] = `"@@MT/TICKETS/SET_RSVP_END_DATE"`;

exports[`RSVP block types SET_RSVP_END_DATE_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_END_DATE_INPUT"`;

exports[`RSVP block types SET_RSVP_END_DATE_MOMENT 1`] = `"@@MT/TICKETS/SET_RSVP_END_DATE_MOMENT"`;

exports[`RSVP block types SET_RSVP_END_TIME 1`] = `"@@MT/TICKETS/SET_RSVP_END_TIME"`;

exports[`RSVP block types SET_RSVP_END_TIME_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_END_TIME_INPUT"`;

exports[`RSVP block types SET_RSVP_GOING_COUNT 1`] = `"@@MT/TICKETS/SET_RSVP_GOING_COUNT"`;

exports[`RSVP block types SET_RSVP_HAS_ATTENDEE_INFO_FIELDS 1`] = `"@@MT/TICKETS/SET_RSVP_HAS_ATTENDEE_INFO_FIELDS"`;

exports[`RSVP block types SET_RSVP_HAS_CHANGES 1`] = `"@@MT/TICKETS/SET_RSVP_HAS_CHANGES"`;

exports[`RSVP block types SET_RSVP_HAS_DURATION_ERROR 1`] = `"@@MT/TICKETS/SET_RSVP_HAS_DURATION_ERROR"`;

exports[`RSVP block types SET_RSVP_HEADER_IMAGE 1`] = `"@@MT/TICKETS/SET_RSVP_HEADER_IMAGE"`;

exports[`RSVP block types SET_RSVP_ID 1`] = `"@@MT/TICKETS/SET_RSVP_ID"`;

exports[`RSVP block types SET_RSVP_IS_LOADING 1`] = `"@@MT/TICKETS/SET_RSVP_IS_LOADING"`;

exports[`RSVP block types SET_RSVP_IS_MODAL_OPEN 1`] = `"@@MT/TICKETS/SET_RSVP_IS_MODAL_OPEN"`;

exports[`RSVP block types SET_RSVP_IS_SETTINGS_LOADING 1`] = `"@@MT/TICKETS/SET_RSVP_IS_SETTINGS_LOADING"`;

exports[`RSVP block types SET_RSVP_NOT_GOING_COUNT 1`] = `"@@MT/TICKETS/SET_RSVP_NOT_GOING_COUNT"`;

exports[`RSVP block types SET_RSVP_NOT_GOING_RESPONSES 1`] = `"@@MT/TICKETS/SET_RSVP_NOT_GOING_RESPONSES"`;

exports[`RSVP block types SET_RSVP_SETTINGS_OPEN 1`] = `"@@MT/TICKETS/SET_RSVP_SETTINGS_OPEN"`;

exports[`RSVP block types SET_RSVP_START_DATE 1`] = `"@@MT/TICKETS/SET_RSVP_START_DATE"`;

exports[`RSVP block types SET_RSVP_START_DATE_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_START_DATE_INPUT"`;

exports[`RSVP block types SET_RSVP_START_DATE_MOMENT 1`] = `"@@MT/TICKETS/SET_RSVP_START_DATE_MOMENT"`;

exports[`RSVP block types SET_RSVP_START_TIME 1`] = `"@@MT/TICKETS/SET_RSVP_START_TIME"`;

exports[`RSVP block types SET_RSVP_START_TIME_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_START_TIME_INPUT"`;

exports[`RSVP block types SET_RSVP_TEMP_CAPACITY 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_CAPACITY"`;

exports[`RSVP block types SET_RSVP_TEMP_DESCRIPTION 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_DESCRIPTION"`;

exports[`RSVP block types SET_RSVP_TEMP_DETAILS 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_DETAILS"`;

exports[`RSVP block types SET_RSVP_TEMP_END_DATE 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_END_DATE"`;

exports[`RSVP block types SET_RSVP_TEMP_END_DATE_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_END_DATE_INPUT"`;

exports[`RSVP block types SET_RSVP_TEMP_END_DATE_MOMENT 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_END_DATE_MOMENT"`;

exports[`RSVP block types SET_RSVP_TEMP_END_TIME 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_END_TIME"`;

exports[`RSVP block types SET_RSVP_TEMP_END_TIME_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_END_TIME_INPUT"`;

exports[`RSVP block types SET_RSVP_TEMP_NOT_GOING_RESPONSES 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_NOT_GOING_RESPONSES"`;

exports[`RSVP block types SET_RSVP_TEMP_START_DATE 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_START_DATE"`;

exports[`RSVP block types SET_RSVP_TEMP_START_DATE_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_START_DATE_INPUT"`;

exports[`RSVP block types SET_RSVP_TEMP_START_DATE_MOMENT 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_START_DATE_MOMENT"`;

exports[`RSVP block types SET_RSVP_TEMP_START_TIME 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_START_TIME"`;

exports[`RSVP block types SET_RSVP_TEMP_START_TIME_INPUT 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_START_TIME_INPUT"`;

exports[`RSVP block types SET_RSVP_TEMP_TITLE 1`] = `"@@MT/TICKETS/SET_RSVP_TEMP_TITLE"`;

exports[`RSVP block types SET_RSVP_TITLE 1`] = `"@@MT/TICKETS/SET_RSVP_TITLE"`;

exports[`RSVP block types UPDATE_RSVP_HEADER_IMAGE 1`] = `"@@MT/TICKETS/UPDATE_RSVP_HEADER_IMAGE"`;
